app/uk/gov/hmrc/breathingspaceifproxy/controller/service/RequestAuth.scala

```diff
@@ -33,15 +34,28 @@ trait RequestAuth extends AuthorisedFunctions with Helpers with Logging {
 
   val authProviders = AuthProviders(PrivilegedApplication)
 
-  def authAction(scope: String): ActionBuilder[Request, AnyContent] =
+  def authAction(scope: String, requestNino: Option[String] = None): ActionBuilder[Request, AnyContent] =
     new ActionBuilder[Request, AnyContent] {
 
       override def parser: BodyParser[AnyContent] = controllerComponents.parsers.defaultBodyParser
       override protected def executionContext: ExecutionContext = controllerComponents.executionContext
 
       override def invokeBlock[A](request: Request[A], f: Request[A] => Future[Result]): Future[Result] = {
+        val notAuthorised = HttpError(retrieveCorrelationId(request), ErrorItem(NOT_AUTHORISED, None)).send
+
+        def checkNino(nino: String): Boolean = requestNino match {
+          case Some(someNino) => someNino == nino
+          case _ => false
+        }
+
         val headerCarrier = HeaderCarrierConverter.fromRequest(request)
-        authorised(authProviders.and(Enrolment(scope)))(f(request))(headerCarrier, executionContext)
+        authorised(authProviders.and(Enrolment(scope)))
```

Comment:
You need tests to cover this code, we will have tests that check how they work together but still some basic tests for each of the scenarios are required.
I would cover
Priviledged application can access any data
Standard application where ninos match and CL200 can access data
Standard application where ninos match trusted helper and CL200 can access data

Standard application with a client id where ninos don't match and CL200 cannot access data
Standard application with no client id where ninos don't match with CL200 cannot access data
Standard application with no client id where ninos don't match with CL200 cannot access data
Standard application where ninos match and CL is not 200 cannot access data